aptible/stack.go

```diff
@@ -0,0 +1,81 @@
+package aptible
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aptible/go-deploy/client/operations"
+)
+
+type Stack struct {
+	ID             int64
+	OrganizationID string
+	Name           string
+}
+
+func (s Stack) isShared() bool {
+	return s.OrganizationID == ""
+}
+
+func (c *Client) GetStacks() ([]Stack, error) {
+	page := int64(0)
+	params := operations.NewGetStacksParams().WithPage(&page)
+
+	stacks, err := c.Client.Operations.GetStacks(params, c.Token)
+	if err != nil {
+		return nil, err
+	}
+
+	stacksToReturn := make([]Stack, len(stacks.Payload.Embedded.Stacks))
+	for idx := range stacksToReturn {
+		stacksToReturn[idx] = Stack{
+			ID:   stacks.Payload.Embedded.Stacks[idx].ID,
+			Name: stacks.Payload.Embedded.Stacks[idx].Name,
+		}
+		if stacks.Payload.Embedded.Stacks[idx].Links.Organization != nil &&
+			stacks.Payload.Embedded.Stacks[idx].Links.Organization.Href != "" {
+			orgIdParts := strings.Split(stacks.Payload.Embedded.Stacks[idx].Links.Organization.Href.String(), "/")
+			stacksToReturn[idx].OrganizationID = orgIdParts[len(orgIdParts)-1]
+		}
+	}
+
+	return stacksToReturn, nil
+}
+
+func (c *Client) GetStackById(id int64) (Stack, error) {
```

Comment:
The ID is implied. This is how the other getters are set up.

```suggestion
func (c *Client) GetStack(id int64) (Stack, error) {
```

Reply:
doing it but i disagree because there's a get stackbyname getstackbyid, get stack alone is agnostic and could mean other things. coding preference, but accepting because it's consistent naming wise